pyproject.toml

```diff
@@ -18,6 +13,38 @@ ignore = [
     "PLR0913", # Too many arguments to function call
     "PLC1901", # `<VAR> != ''` can be simplified to `<VAR>` as an empty string is falsey
     "PLR2004", # Magic value used in comparison
+    # FIXME stop ignoring these and fix them (see https://github.com/hail-is/hail/pull/14131)
+    "I001",    # Import block is un-sorted or un-formatted
+    "F405",    # <VAR> may be undefined, or defined from star imports
+    "E712",    # Comparison to `<True|False>` should be `cond is <True|False>`
+    "PLW2901", # `with` statement variable overwritten by assignment target
+    "F841",    # Local variable is assigned to but never used
+    "RUF012",  # Mutable class attributes should be annotated with `typing.ClassVar`
+    "PLW2901", # `for` loop variable overwritten by assignment target
+    "PLR5501", # Use `elif` instead of `else` then `if`, to reduce indentation
+    "PLR1714", # Consider merging multiple comparisons: `<VAL> in (<VAL1>, <VAL2>)`.
+    "F541",    # f-string without any placeholders
+    "RUF010",  # Use explicit conversion flag
+    "PLR0402", # Use `from <MODULE> import <VAL>` in lieu of alias
+    "RUF005",  # Consider `[*<LIST1>, *<LIST2>]` instead of concatenation
+    "E711",    # Comparison to `None` should be `cond is None`
+    "F403",    # Star import used; unable to detect undefined names
+    "F401",    # Import imported but unused
+    "PLW3301", # Nested `max` calls can be flattened
+    "RUF013",  # PEP 484 prohibits implicit `Optional`
+    "PLR1701", # Merge `isinstance` calls
+    "RUF100",  # Unused `noqa` directive
+    "RUF015",  # Prefer `next(...)` over single element slice
+    "F811",    # Redefinition of unused <VAR>
+    "F601",    # Dictionary key literal repeated
+    "PLR0124", # Name compared with itself
+    "E402",    # Module level import not at top of file
+    "PLW0602", # Using global but no assignment is done
+    "RUF002",  # Docstring contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?
+    "PLC0132", # `TypeVar` name does not match assigned variable name
+    "PLE0307", # `__str__` does not return `str`
+    "PLC0414", # Import alias does not rename original package
+    "PLW0127", # Self-assignment of variable
```

Comment:
I am all for these changes, but IIUC these are related to expanding the scope of ruff as a linter to include the `hail` directory, and not necessarily to do with replacing black as a formatter. I'm happy to review a big PR that includes linting fixes, just as long as it doesn't include auto-formatting changes.